### PR TITLE
ACCUMULO-4600: Fix to properly read from accumulo-site.xml

### PR DESCRIPTION
**Pull request by milleruntime**
Changes to Shell that should fix this problem without having to revert ACCUMULO-4505

TODO: Fix ShellConfigTest to properly test and not just compile.

### CODE REVIEW

File: shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void testZooKeeperHostFallBackToSite() throws Exception {
     Map<String,String> data = new HashMap<>();
     data.put(Property.INSTANCE_ZK_HOST.getKey(), "site_hostname");
     AccumuloConfiguration conf = new ConfigurationCopy(data);
```
Comment (keith-turner) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: Seems like this configuration is being created and not used.  Does this test pass?

Comment (milleruntime) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: No see above TODO

File: shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java [hunk 2]
```diff
@@ -123,29 +118,21 @@ public void testTokenAndOptionAndPassword() throws IOException {
   @Test
   public void testZooKeeperHostFallBackToSite() throws Exception {
     ClientConfiguration clientConfig = new ClientConfiguration();
-    Map<String,String> data = new HashMap<>();
-    data.put(Property.INSTANCE_ZK_HOST.getKey(), "site_hostname");
-    AccumuloConfiguration conf = new ConfigurationCopy(data);
-    assertEquals("site_hostname", Shell.getZooKeepers(null, clientConfig));
+    assertFalse("Client config contains zk hosts", clientConfig.containsKey(ClientConfiguration.ClientProperty.INSTANCE_ZK_HOST.getKey()));
+    assertEquals("localhost:2181", Shell.getZooKeepers(null, clientConfig));
```
Comment (keith-turner) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: it does not seem like this is testing falling back to site

Comment (milleruntime) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: It does not seem that way but the last resort of `Shell.getZooKeepers()` is to make a static call to get the site configuration:
`SiteConfiguration.getInstance(ClientContext.convertClientConfig(clientConfig)).get(Property.INSTANCE_ZK_HOST)`

Comment (keith-turner) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: So its testing that it falls back to the default?

Comment (milleruntime) [inline shell/src/test/java/org/apache/accumulo/shell/ShellConfigTest.java]: Yeah it was.  I added the site file to resources in 2b57295

**Pull request merged**